style-sync/src/main/scala/commercialtools/stylesync/StyleSync.scala

```diff
@@ -0,0 +1,146 @@
+import java.nio.file.Paths
+
+import com.google.api.ads.dfp.axis.v201708.{CreativeTemplate, CreativeTemplateType, NativeStyle}
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.io.Source
+
+trait Logging {
+  val logger: Logger = LoggerFactory.getLogger(getClass)
+}
+
+case class GuStyle(formatName: String,
+                   dfpFormat: Option[CreativeTemplate],
+                   styleName: String,
+                   dfpStyle: Option[NativeStyle],
+                   localPath: String,
+                   testFile: Option[File]) {
+
+  val localHtmlLines: List[String] = Source.fromFile(Paths.get(localPath, "index.html").toFile).getLines.toList
+  val localCssLines: List[String] = Source.fromFile(Paths.get(localPath, "index.css").toFile).getLines.toList
+}
+
+object StyleSync extends App with Logging {
+
+  /*
+     Builds up a list of GuStyle objects, which link locally built templates to their
+     respective formats (CreativeTemplates) and styles (NativeStyles) in DFP.
+  */
+  def findStyles(buildRootDirectory: File,
+                 nativeFormats: => List[CreativeTemplate],
+                 nativeStyles: => List[NativeStyle]): List[GuStyle] = {
+
+    def titleCase(str: String, delim: String = " "): String =
+      str.split(delim).map(word =>
+        word.head.toUpper + word.tail.toLowerCase
+      ).mkString(delim)
+
+    def deriveFormatName(formatDirectory: String) =
+      titleCase(formatDirectory.replaceAll("-", " "))
+      .replaceAll("Capi", "cAPI")
+
+    def deriveStyleName(styleDirectory: String, formatName: String) = {
+      val platform = if(styleDirectory == "amp") "AMP" else styleDirectory.capitalize
+      s"$platform - $formatName"
+    }
+
+    for {
+      formatDirectory <- buildRootDirectory.listFiles.toList
+      if formatDirectory.isDirectory
+      styleDirectory <- formatDirectory.listFiles.toList
+      if styleDirectory.isDirectory
+    } yield {
+
+      val formatName: String = deriveFormatName(formatDirectory.getName)
+      val styleName: String = deriveStyleName(styleDirectory.getName, formatName)
+
+      val format: Option[CreativeTemplate] = nativeFormats find { _.getName == formatName }
+      val style: Option[NativeStyle] = nativeStyles find { _.getName == styleName}
+      val testFile: Option[File] = formatDirectory.listFiles.find(_.getName == "test.json")
+
+      GuStyle(
+              formatName = formatName,
+              dfpFormat = format,
+              styleName = styleName,
+              dfpStyle = style,
+              localPath = styleDirectory.getAbsolutePath,
+              testFile)
+    }
+  }
+
+  case class DiffLine(line: String, number: Int)
+  case class DiffResult(diffLines: List[DiffLine], sourceALines: List[String], sourceBLines: List[String])
+
+  def summarise(formats: List[CreativeTemplate], styles: List[NativeStyle], guStyles: List[GuStyle]) = {
+
+    val divider: String = "------------------------------------------------------------------"
```

Comment:
It's good practice to declare constants on Object so that they aren't initialised every time the function is invoked.